### PR TITLE
Aria-label that indicates scrolling behaviour interferes in headings content for screen readers

### PR DESCRIPTION
**Pull request by marinaaisa**
Bug/issue #111006953, if applicable: 

## Summary

Aria-label that indicates scrolling behaviour interferes in headings content for screen readers

It's been fixed by restructuring how the anchor link is set and adding AX description that should read: Content of heading + "in page link" to describe that the link is within the page.

Screenshot 2023-06-26 at 20 30 31

## Dependencies

NA

## Testing

Steps:
1. Run any doccarchive
2. Go to section title
3. Assert that you can read its content using VoiceOver

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by marinaaisa** [comment]
@swift-ci test

File: src/components/ContentNode/LinkableHeading.vue [hunk 1]
```diff
@@ -9,24 +9,25 @@
 -->
 
 <template>
-  <component
-    :id="anchor"
-    :is="`h${level}`"
-  >
+  <div class="heading-wrapper">
```
Comment (mportiz08) [inline src/components/ContentNode/LinkableHeading.vue:12]: One downside to this restructuring of elements is that now you have to click the actual icon in order to get the link for the heading whereas before you could click anywhere within the title itself, which seemed like a better experience to me.

Is there a way of keeping that same behavior as before while also addressing the screen reader issue?

File: src/components/ContentNode/LinkableHeading.vue [hunk 2]
```diff
@@ -9,24 +9,25 @@
...
       v-if="shouldLink"
       :to="{ hash: `#${anchor}` }"
-      class="header-anchor"
-      aria-label="Scroll to section"
+      class="anchor"
       @click="handleFocusAndScroll(anchor)"
     >
-      <slot />
       <LinkIcon class="icon" aria-hidden="true" />
+      <span :aria-labelledby="anchor" class="visuallyhidden" />
```
Comment (dobromir-hristov) [inline src/components/ContentNode/LinkableHeading.vue:27]: cant we just put an aria-label on the link?

**Review by dobromir-hristov — COMMENTED** [review]
Change looks good, its a bit close to how Vue docs render, apart from the position of the anchor itself.

File: src/components/ContentNode/LinkableHeading.vue (hunk 1 above)
Comment (marinaaisa) [reply src/components/ContentNode/LinkableHeading.vue]: I took another approach and I continued with the previous experience. So the change isn't too much https://github.com/apple/swift-docc-render/pull/698/commits/f9ce461bd83c90d5331108648b90b58d171abcf3

File: src/components/ContentNode/LinkableHeading.vue (hunk 2 above)
Comment (marinaaisa) [reply src/components/ContentNode/LinkableHeading.vue]: I decided to keep the previous style fixing the AX issue https://github.com/apple/swift-docc-render/pull/698/commits/f9ce461bd83c90d5331108648b90b58d171abcf3

File: src/components/ContentNode/LinkableHeading.vue [hunk 3]
```diff
@@ -17,10 +17,10 @@
       v-if="shouldLink"
       :to="{ hash: `#${anchor}` }"
       class="header-anchor"
-      aria-label="Scroll to section"
       @click="handleFocusAndScroll(anchor)"
     >
       <slot />
+      <span class="visuallyhidden" >{{ $t('accessibility.in-page-link') }}</span>
```
Comment (mportiz08) [inline src/components/ContentNode/LinkableHeading.vue:23]: Works great, but I think you'll need to update the unit tests for this change (they're failing now)

**Review by mportiz08 — CHANGES_REQUESTED** [review]
Simpler approach works great 🚀

The tests are failing now though and need to be updated.

File: src/components/ContentNode/LinkableHeading.vue (hunk 3 above)
Comment (marinaaisa) [reply src/components/ContentNode/LinkableHeading.vue:23]: Oh yeah, of course 🤦‍♀️  https://github.com/apple/swift-docc-render/pull/698/commits/8259023f76d5a62b9ba3ed7b7214fa9cf2756ba0

**Comment by marinaaisa** [comment]
@swift-ci test

**Review by mportiz08 — APPROVED** [review]

**Pull request merged**